uaclient/apt_news.py

```diff
@@ -90,6 +103,41 @@ def do_selectors_apply(
         if selectors.pro != _is_attached(cfg).is_attached:
             return False
 
+    if selectors.architectures is not None:
+        if system.get_dpkg_arch() not in selectors.architectures:
+            return False
+
+    if selectors.packages is not None:
+        for package in selectors.packages:
+            try:
+                package_name, version_operator, package_version = package
+            except ValueError:
+                LOG.warning("Invalid package selector: %r", package)
+                return False
+            if not is_installed(package_name):
+                LOG.info("Package %s is not installed", package_name)
+                return False
+            version_comparison = version_compare(
+                get_pkg_version(package_name), package_version
```

Comment:
I think we need to think of the situation where this function returns `None` (when the package doesn't exist yet)

Reply:
Would that not be flagged be the installed package check and return False earlier?

Reply:
Yes, you are totally right. But now, looking at the code, I think it might be better if we don't have that call to `is_installed` at all.

My rationale here is that this function will fetch all installed packages and then perform a search on that list.
For the `get_pkg_version` we directly check the package in the apt cache, which should be faster.

We would then return false only if `get_pkg_version` returns `None`.

What do you think ?

Reply:
Oh yes, actually, that makes a lot of sense. That would be a good change, thanks @lucasmoura